src/components/MAPI/apps/AppInstallModal.tsx

```diff
@@ -0,0 +1,393 @@
+import ClusterPicker from 'Apps/AppDetail/InstallAppModal/ClusterPicker';
+import InstallAppForm from 'Apps/AppDetail/InstallAppModal/InstallAppForm';
+import { validateAppName } from 'Apps/AppDetail/InstallAppModal/utils';
+import { useAuthProvider } from 'Auth/MAPI/MapiAuthProvider';
+import GenericModal from 'components/Modals/GenericModal';
+import { push } from 'connected-react-router';
+import yaml from 'js-yaml';
+import { FlashMessage, messageTTL, messageType } from 'lib/flashMessage';
+import useDebounce from 'lib/hooks/useDebounce';
+import { useHttpClientFactory } from 'lib/hooks/useHttpClientFactory';
+import RoutePath from 'lib/routePath';
+import { GenericResponse } from 'model/clients/GenericResponse';
+import * as capiv1alpha3 from 'model/services/mapi/capiv1alpha3';
+import * as metav1 from 'model/services/mapi/metav1';
+import PropTypes from 'prop-types';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { OrganizationsRoutes } from 'shared/constants/routes';
+import { IAsynchronousDispatch } from 'stores/asynchronousAction';
+import { IState } from 'stores/state';
+import useSWR from 'swr';
+import Button from 'UI/Controls/Button';
+import ClusterIDLabel from 'UI/Display/Cluster/ClusterIDLabel';
+
+import { createApp, filterClusters } from './utils';
+
+function mapClusterToClusterPickerInput(
+  cluster: capiv1alpha3.ICluster
+): React.ComponentPropsWithoutRef<typeof ClusterPicker>['clusters'][0] {
+  return {
+    id: cluster.metadata.name,
+    name: capiv1alpha3.getClusterDescription(cluster),
+    owner: capiv1alpha3.getClusterOrganization(cluster) ?? '',
+    isAvailable: true,
+  };
+}
+
+const SEARCH_DEBOUNCE_RATE = 250; // In ms.
+
+const CLUSTER_PICKER_PAGE = 0;
+const APP_FORM_PAGE = 1;
+const pages: ReadonlyArray<number> = [CLUSTER_PICKER_PAGE, APP_FORM_PAGE];
+
+interface IAppInstallModalApp {
+  catalog: string;
+  name: string;
+  versions: IAppCatalogAppVersion[];
+}
+
+interface IAppInstallModalProps {
+  app: IAppInstallModalApp;
+  selectedClusterID: string | null;
+}
+
+const AppInstallModal: React.FC<IAppInstallModalProps> = (props) => {
```

Comment:
I duplicated the old `InstallAppModal` component, and replaced the REST API functionality with the MAPI one.